### ISSUE TITLE
Accessibility issue with thousand separator as " "

### ISSUE THREAD

**Issue by sarujo** [s-yadav/react-number-format#540]
#### Describe the issue and the actual behaviour
If the thousand separator is passed in as "." or "," then the number is read as a whole (1.000 = one thousand), but if thousand separator is passed in as space (" ") it is read as as separate numbers (1 000 = one zero zero zero, or 22 444 = twenty two, four hundred and forty four)

#### Describe the expected behaviour
Empty space as a separator should work the same way as other common separators (such as "." or ",") and not make a screen reader to think that it separates numbers.

#### Provide a CodeSandbox link illustrating the issue
<NumberFormat
	  id={id}
	  name={name}
	  value={value}
	  thousandSeparator={" "}
	  {...ariaProps}
	  {...rest}
/>
#### Provide steps to reproduce this issue
- Add a thousandSeparator={" "} to NumberFormat component, type in number and use screen reader to check the result.

#### Please check the browsers where the issue is seen
- [ ] Chrome

**Comment by nikhil-varma** [comment]
Hey! Thanks for reporting this issue. We will investigate and shoot out a resolution soon. Keep an eye out for updates 🙂

**Comment by s-yadav** [comment]
This is one classic issue with masking/formatting input values. As all the formatting is happening inside an input tag, we don't have much control on how the input value should be treated by screen readers. Not sure how this can be fixed on the library, but open for any approach if anyone knows.

One approach people fix this is by formatting characters only on blur, and on focus remove all formatting characters on focus. This lib does it (https://github.com/filamentgroup/politespace). But this will completely break the expectation people using this lib. May be we can provide some prop for that. 

But let's discuss if you know of any other solution.
